sparkly/functions.py

```diff
@@ -107,6 +113,18 @@ def switch_case(switch, case=None, default=None, **additional_cases):
         ).when(
         ... F.col('state') == 'NY', 'New York'
         ).otherwise('Other')
+
+        If you need to "bucketize" a value
+
+        ``switch_case('age', {(13, 17): 1, (18, 24): 2, ...}, operand=lambda c, v: c.between(*v))``
+
+        is equivalent to
+
+        >>> F.when(
```

Comment:
Honestly I like the pyspark's variant much more, it's much less cryptic and easy to read. You are just saving a bit of typing, but KO says your write once and read multiple times. 
So imo at least signature of this method needs some nice "dsl".

Reply:
Hi @skliarpawlo. Thanks for the feedback - is there anything specific you'd change from the function signature to make it easier to understand / use? Also consider that this PR just adds one optional argument (`operand`) in a backwards compatible improvement, so introducing a completely new dsl would be kinda out of scope here.

Having said that, Scott provides a good usecase above. For example, think of a case where you have a continuous age column and you want to transform it into discrete, pre-defined buckets. Then your configuration might specify something like
```
AGE_BUCKETS = [(13, 17), (18, 24), (25, 34), ...]
```
In your code, this allows you to say
```
switch_case(
    'age', 
    case={bucket_limits: '_'.join(bucket_limits) for bucket_limits in AGE_BUCKETS}, 
    operand=lambda c, v: c.between(*v),
)
```
which is very compact (and straightforward once you get familiar with what the arguments are). It is way more difficult to chain `when` statements together that way, especially when you can't list explicitly all the cases. 

Even when you can list explicitly all cases though, I've found `switch_case` to be much more convenient for readability purposes in the usual case, as it omits a lot of the boilerplate and lists the cases directly for the reader to skim over. But that's just a personal preference.

Reply:
I'd say it's compact but it's not easy for an eye. The same or even better from point for familiarity is code like (I'm sure there are mistakes but you got the point):
```
reduce(lambda x, y: x.when(*y), [
		(F.col('age').between(*group), F.lit(val))
	] for group, val in AGE_BUCKETS.items(), F)
```

I agree that adding more pyspark-like dsl'y stuff will take time and making the already existing function more general is a good thing so LGTM.